lib/datasource/index.ts

```diff
@@ -52,7 +53,25 @@ async function getRegistryReleases(
   config: GetReleasesConfig,
   registryUrl: string
 ): Promise<ReleaseResult> {
+  const cacheKey = `${datasource.id} ${registryUrl} ${config.lookupName}`;
+  if (datasource.caching) {
+    const cachedResult = await packageCache.get<ReleaseResult>(
+      cacheNamespace,
+      cacheKey
+    );
+    // istanbul ignore if
+    if (cachedResult) {
+      logger.debug({ cacheKey }, 'Returning cached datasource response');
+      return cachedResult;
+    }
+  }
   const res = await datasource.getReleases({ ...config, registryUrl });
+  // cache non-null responses unless marked as private
```

Comment:
why only non-null?

Reply:
Trying to avoid blocking a private package with wrong or missing credentials being blocked by someone with correct credentials. 

Ie null can mean "private package without proper credentials"